src/main/java/cz/kominekjan/disenchantment/guis/impl/EnchantmentsGUI.java

```diff
@@ -91,51 +93,46 @@ public final GUIItem[] getEnchantmentsItems(List<Enchantment> enchantments) {
+            String lore2 = bookStatus.getDisplayName() + " " + ChatColor.GRAY + "for " + ChatColor.LIGHT_PURPLE + ChatColor.BOLD + "Book Splitting";
 
             worldItems[i] = new GUIItem(
                     freeSlots[i],
-                    DefaultGUIElements.enchantmentItem(ChatColor.LIGHT_PURPLE + "" + ChatColor.BOLD + enchantment.getKey().getKey(), lore),
+                    DefaultGUIElements.enchantmentItem(ChatColor.LIGHT_PURPLE + "" + ChatColor.BOLD + enchantment.getKey().getKey(), lore, lore2),
                     event -> {
                         event.setCancelled(true);
 
-                        Map<Enchantment, Boolean> disabledEnchantments = getDisabledEnchantments();
-                        Map.Entry<Enchantment, Boolean> disabledEnchantment = disabledEnchantments.entrySet().stream().filter(e -> e.getKey().equals(enchantment)).findFirst().orElse(null);
-                        String newLore = "";
-
-                        if (disabledEnchantment != null) {
-                            if (disabledEnchantment.getValue()) {
-                                disabledEnchantments.remove(disabledEnchantment.getKey());
-                                disabledEnchantments.put(enchantment, false);
-                            } else {
-                                disabledEnchantments.remove(disabledEnchantment.getKey());
-                            }
-
-                            newLore = disabledEnchantment.getValue() ? ChatColor.RED + "Cancel" : ChatColor.GREEN + "Enabled";
-                        } else {
-                            disabledEnchantments.put(enchantment, true);
-                            newLore = ChatColor.GOLD + "Keep";
-                        }
+                        boolean bookSplitting = event.isRightClick();
+
+                        EnchantmentStatus oldStatus = getEnchantmentStatus(enchantment);
+                        EnchantmentStatus oldBookStatus = getBookSplittingEnchantmentStatus(enchantment);
 
-                        setDisabledEnchantments(disabledEnchantments);
+                        EnchantmentStatus newStatus = bookSplitting ? oldStatus : EnchantmentStatus.getNextStatus(oldStatus);
+                        EnchantmentStatus bookNewStatus = bookSplitting ? EnchantmentStatus.getNextStatus(oldBookStatus) : oldBookStatus;
+
+                        String newLore  = newStatus.getDisplayName() + " " + ChatColor.GRAY + "for " + ChatColor.LIGHT_PURPLE + ChatColor.BOLD + "Disenchantment";
+                        String newLore2 = bookNewStatus.getDisplayName() + " " + ChatColor.GRAY + "for " + ChatColor.LIGHT_PURPLE + ChatColor.BOLD + "Book Splitting";
+
+                        if(bookSplitting){
+                            setBookSplittingEnchantmentsStatus(enchantment, bookNewStatus);
+                        }else{
+                            setEnchantmentsStatus(enchantment, newStatus);
+                        }
```

Comment:
While it is true that it would make a wasteful read. I think having up-to-date state of the other status would be better as:
1. This is an admin side of the plugin and would probably be rarely used. So small performance improvement is less likely to impact anything at all. Especially since (from what I know) config is stored in ram (no disk read).
2. It is not unlikely that multiple admin try to edit the same configuration (for example, when they just installed the plugin). Reason why up-to-date value on click would be better in my opinion.

But I guess we could only depend on previous state and do 0 config read on click. That would make using the GUI more sens when multiple admin use it at the same time and avoid strange behavior caused by outdated view by modification by other admin.

Reply:
Probably resolved with https://github.com/H7KZ/Disenchantment/pull/15/commits/be329a3a35cfd14a12c38932cd9332b56b2ba090. Please check if the new code is okay for you.

Reply:
Great idea with the previous state! Didn't even think of it! Nice!!